Source/Urho3D/Math/Rect.h

```diff
@@ -181,123 +192,53 @@ class URHO3D_API Rect
             return INSIDE;
     }
 
-    /// Return float data.
-    const void* Data() const { return &min_.x_; }
+    /// Return number data.
+    const T* Data() const { return &min_.x_; }
 
     /// Return as a vector.
-    Vector4 ToVector4() const { return Vector4(min_.x_, min_.y_, max_.x_, max_.y_); }
+    BaseVector4<T> ToVector4() const { return BaseVector4<T>(min_.x_, min_.y_, max_.x_, max_.y_); }
 
     /// Return as string.
     String ToString() const;
 
-    /// Minimum vector.
-    Vector2 min_;
-    /// Maximum vector.
-    Vector2 max_;
+    union
+    {
+        /// Minimum vector.
+        BaseVector2<T> min_;
+        struct

```

Comment:
There is no anonymous structs in C++

Reply:
Compilers tend to support it. This was also used out of necessity, in order to avoid change rippling through big part of the codebase. Do you have a better idea how this could be solved?

Reply:
It somehow worked without anonymous unnamed structs before, so I just thought it could continue working...

Reply:
Yeah see issue is that `Rect` was defined as `Vector2 min_; Vector2 max_`, but `IntRect` was defined as `int left_; int top_; int right_; int bottom_;`. I used anonymous structs here in order to maintain backwards compatibility with `IntRect`. If we were to throw anonymous structs out then i would have to go through code of all engine and change `.left_` to `.min.x_` and so on. Do we really want that?